### ISSUE TITLE
permitted-contents: transparent elements bypass parent content-model checks

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#3928]
## Problem

`permitted-contents` incorrectly ignores the identity of transparent-content-model elements when validating their placement in a parent's content model. It flattens the transparent element to its children before running the parent's selector, dropping the transparent element itself from the check. As a result, cases where the transparent element itself is invalid in the parent's content model go undetected.

## Repro

```html
<button><a href="/x"><span>text</span></a></button>
```

Expected: `permitted-contents` fires because `<a href>` is interactive content and `<button>` disallows interactive descendants (HTML LS §4.10.9 button).

Observed: passes (0 violations).

Same shape reproduces for:

- `<button><audio controls></audio></button>` (audio is transparent when there is an `src` attribute, otherwise has a partly-transparent tail; transparent unwrapping still skips the element)
- `<picture><video></video><img src=x alt></picture>` (video is transparent, picture rejects video)
- `<picture><audio></audio><img src=x alt></picture>`
- `<picture><canvas></canvas><img src=x alt></picture>`
- `<picture><a><img src=x alt></a></picture>`
- `<picture><ins></ins><img src=x alt></picture>`

Non-transparent counterparts DO fire correctly, confirming the bug is scoped to the transparent-flattening path:

- `<button><input type=text></button>` — invalid-attr fires (input is not transparent, not phrasing)
- `<button><label>lbl</label></button>` — fires
- `<picture><p></p><img src=x alt></picture>` — fires
- `<picture><embed src=x><img src=x alt></picture>` — fires
- `<button><span><a href></a></span></button>` — fires (a is nested inside a non-transparent span)

## Root cause

`packages/@markuplint/rules/src/permitted-contents/represent-transparent-nodes.ts` replaces every transparent-model child with its own children (an empty list if the transparent element is empty). The parent's content-model selector then runs on the flattened list, which no longer contains the transparent element. Any child-level constraint on that element (e.g. `:not(:model(interactive))`) never gets evaluated.

## Impact on bench

Directly blocks the following nu-only fixtures from flipping to `match-error`:

- `html/elements/a/with-href-button-descendant-novalid.html` — `<button><a href></a></button>`
- `html/elements/audio/controls-in-button-novalid.html` — `<button><audio controls></audio></button>`
- `html/elements/picture/junk-noscript-novalid.html`
- `html/elements/picture/junk-noscript-after-source-no-img-novalid.html`
- `html/elements/picture/junk-video-before-novalid.html`

## Spec basis

- HTML LS §3.2.5.3 [Transparent content models](https://html.spec.whatwg.org/multipage/dom.html#transparent-content-models): the transparent portion of an element's content model defers *the constraints on the element's children* to the parent's content model; it does NOT relax the parent's own constraints on where the transparent element itself may appear.
- HTML LS §4.10.9 [the button element](https://html.spec.whatwg.org/multipage/form-elements.html#the-button-element): "Content model: Phrasing content, but there must be no interactive content descendant and no descendant with the tabindex attribute specified."
- HTML LS §4.8.1 [the picture element](https://html.spec.whatwg.org/multipage/embedded-content.html#the-picture-element): "Content model: Zero or more source elements, followed by one img element, optionally intermixed with script-supporting elements."

## Proposed fix (sketch — needs design)

`representTransparentNodes` should keep the transparent element in the flattened list alongside its children. The parent's selector then evaluates both the element (for its own placement) and the children (for the transparent inheritance). The `oneOrMore` counting logic already treats each child independently, so the child-count semantics do not change; only the "invisible transparent" bug goes away.

Risk: the existing 23+ transparent tests need to be re-checked against this change. Regression surface should be measured before landing.

## Related fixtures already flipped

Not blocked by this bug; kept for reference:

- `srcset-sizes-constraint` (PR #3888/#3909) handled picture edge cases without touching the transparent bug.

## Benchmark cross-reference

_Sync'd from `tests/external/snapshots/` (submodule=1429313, nu-validator@sha256:210bbc353b23, markuplint=5.0.0-rc.4, generated=2026-08-12 03:50:09 UTC)._

Filter: `^html\/elements\/(a\/with-href-button-descendant|audio\/controls-in-button|picture\/(junk-noscript|junk-noscript-after-source-no-img|junk-video-before))-novalid` → **5** fixtures. Verdict tally: match-error=5, match-clean=0, ml-only=0, nu-only=0, nu-over=0.

Fixed. `representTransparentNodes` now keeps the transparent element itself in the flattened list alongside its pass-through children, so the parent's own content-model check still evaluates the wrapper's placement (HTML LS §3.2.5.3: transparency defers only the *children* to the parent's model; the transparent element itself must still satisfy the parent's constraints). All 5 fixtures now `match-error`; verified against the full 5442-fixture corpus with zero ml-only/nu-over regressions (path sets identical before/after). Safe to close.

| fixture | verdict | nu id / ml rule |
| --- | --- | --- |
| `html/elements/a/with-href-button-descendant-novalid.html` | match-error | — |
| `html/elements/audio/controls-in-button-novalid.html` | match-error | — |
| `html/elements/picture/junk-noscript-after-source-no-img-novalid.html` | match-error | — |
| `html/elements/picture/junk-noscript-novalid.html` | match-error | — |
| `html/elements/picture/junk-video-before-novalid.html` | match-error | — |

Reproduce: see [`bench-triage` skill](https://github.com/markuplint/markuplint/blob/dev/.claude/skills/bench-triage/SKILL.md) for the audit workflow.
